File: example/jchatbot/src/main/java/zw/co/dcl/jchatbot/configs/SessionLocator.java
package zw.co.dcl.jchatbot.configs;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;
import zw.co.dcl.jawce.session.ISessionManager;

/**
 * Session service locator
 * <p>
 * Used on non-spring service classes (usually Hook classes)
 * <p>
 * to get access to ISessionManager bean
 */
@Component
public class SessionLocator {
    private static ISessionManager sessionManager;

    public static ISessionManager getSessionManager() {
        if(sessionManager == null) {
            throw new IllegalStateException("SessionManager not initialized.");
        }
        return sessionManager;
    }

    @Autowired
    public void setSessionManager(SessionConfig sessionConfig) {
        SessionLocator.sessionManager = sessionConfig.sessionManager();
    }
}

File: chatbot/src/main/kotlin/zw/co/dcl/jawce/chatbot/configs/SessionLocator.kt
package zw.co.dcl.jawce.chatbot.configs

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component
import zw.co.dcl.jawce.session.ISessionManager

@Component
class SessionLocator {
    companion object {
        var sessionManager: ISessionManager? = null
    }

    @Autowired
    fun setISessionManager(sessionConfig: SessionConfig) {
        sessionManager = sessionConfig.sessionManager()
    }
}
